customers.go

```diff
@@ -169,6 +183,31 @@ func (c customerRepository) Update(customer shopify.Customer) (shopify.Customer,
 	return response.Customer.ToShopify(), nil
 }
 
+func (c customerRepository) GetByQuery(fields []string, query shopify.CustomerSearchQuery) (shopify.Customers, error) {
+	url := c.createURL(fmt.Sprintf("customers/search.json?fields=%v&query=%s", strings.Join(fields, ","), query.String()))
```

Comment:
Fields does not look like it is required in the shopofy documentation? What happens when you pass an empty fields param?

Reply:
It just returns all the fields if it's empty, like it does when we don't give any values to our `fields` query param for the GET mesages

Reply:
For the above comment, you can see the result by hitting this:
`https://lpod-staged.myshopify.com/admin/api/2021-04/customers/search.json?fields=&query=customer_tag:patient_id:1818885`

Reply:
This is perfectly fine then :+1: